### PR TITLE
feat: Attach aws Volume (#1512)

### PR DESCRIPTION
**Pull request by HURUIZHE**
Signed-off-by: <5538343986@qq.com>

Create/Delete Volume complete

### CODE REVIEW

File: controllers/infra/api/v1/infra_types.go [hunk 1]
```diff
@@ -91,6 +92,8 @@ type Hosts struct {
 	Metadata []Metadata `json:"metadata,omitempty"`
 	// Find the mapping between expected hosts and actual hosts
 	Index int `json:"index,omitempty"`
+	// Availability Zone
+	AvailabilityZone string `json:"availability_zone,omitempty"`
```
Comment (fanux) [inline controllers/infra/api/v1/infra_types.go:96]: Seems all the hosts will in the same zone? If the zone is same, put this failed in `Spec.AvailabilityZone` is better?

File: controllers/infra/drivers/aws/get_instance.go [hunk 2]
```diff
@@ -180,34 +185,109 @@ func (d Driver) getInstances(infra *v1.Infra) ([]v1.Hosts, error) {
...
 
+			metadata := v1.Metadata{
+				IP:     []string{*i.PrivateIpAddress},
+				ID:     *i.InstanceId,
+				DiskID: disksID,
+			}
 			if h, ok := hostmap[index]; ok {
 				h.Count++
+				for _, blockDeviceMap := range i.BlockDeviceMappings {
+					if _, ok := diskMap[*blockDeviceMap.Ebs.VolumeId]; !ok {
```
Comment (fanux) [inline controllers/infra/drivers/aws/get_instance.go:204]: Too much indentation can be encapsulated in a function.

File: controllers/infra/drivers/driver.go [hunk 3]
```diff
@@ -30,10 +30,20 @@ import (
...
+	StopInstances(hosts *v1.Hosts) error
+	ModifyInstances(curHosts *v1.Hosts, desHosts *v1.Hosts) error
 	DeleteInstanceByID(instanceID string, infra *v1.Infra) error
 	GetInstancesByLabel(key string, value string, infra *v1.Infra) (*v1.Hosts, error)
 	// get infra all current hosts
 	GetInstances(infra *v1.Infra) ([]v1.Hosts, error)
+	// Volumes operation
+	// Create and Attach
+	CreateVolume(infra *v1.Infra, host *v1.Hosts, disks *v1.Disk) error
+	CreateVolumes(infra *v1.Infra, host *v1.Hosts, disks []v1.Disk) error
```
Comment (fanux) [inline controllers/infra/drivers/driver.go:42]: Seems only need one of those functions. `CreateVolume` is `CreateVolume(xx,xx, 1 volume)`

Comment (HURUIZHE) [reply controllers/infra/drivers/driver.go:42]: CreateVolumes create len(disks) goroutines to run CreateVolume. I think CreateVolume shouldn't be exposed.

Comment (fanux) [reply controllers/infra/drivers/driver.go:41]: yes

**Review by fanux — APPROVED** [review]

**Pull request merged**